app/uk/gov/hmrc/gform/views/form/snippets/uploaded_files.scala.html

```diff
@@ -33,16 +33,10 @@
 @defining(formTemplateId.value, maybeAccessCode.getOrElse("-")) { case (templateId, accessCode) =>
   <div id="@formComponentId-files" class="govuk-body uploaded-files">
     @maybeCurrentValue.map { currentValue =>
-      <span>@{currentValue.replace(fileId.value + "_", "")}</span>
-      <a
-        href="@deleteUrl"
-        class="govuk-link"
-        data-form-component-id="@formComponentId.value"
-        data-form-id="@templateId"
-        data-access-code="@accessCode">
-        <span aria-hidden="true">@messages("file.delete")</span>
-        <span class="govuk-visually-hidden">@messages("file.delete.file", currentValue)</span>
-      </a>
+      <span>@{currentValue.replace(fileId.value + "_", "")} @govukButton(Button(content = Text(messages("file.delete")),
+          inputType=Some("submit"),
+          attributes = Map("id" -> "fileDelete", "formaction" -> uk.gov.hmrc.gform.fileupload.routes.FileUploadController.deleteFile(formTemplateId, maybeAccessCode, formComponentId).path),
+          classes="govuk-button link"))</span>
```

Comment:
This seems to be affecting accessibility. There is no information what file is this delete button removing. Or do I miss something?

Reply:
Thanks. Fixed both in template and javascript using aria-label